### PR TITLE
Allow php.ini changes for all variables

### PR DESCRIPTION
**Pull request by juanmrad**
### Issue:
While using this php image in order to change php ini settings it isn't possible other than the currently 2 supported. This causes issues as php has many configurations which would be handy to change on different environments or be able to change with a simple command over them being set by modifying the php.ini script directly.

### Solution:
This PR will solve this issue by allowing the use of prefixed environment variables `PHP` to ensure that these are for changing php.ini variables only and then adding the extra prefixes `CLI` `FPM` and `ALL` to allow these variables to be modified in CLI, FPM or both environments.
In order to accomplish changes on variables that might have dots `.` on them such as `session.save_path` the variable name should have a double underscore `__` that will be replaced for a dot when parsing the file.

- allow changes of any php.ini variables. by using prefixes on the env variables set to the container
- update readme file to reflect changes.

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -1,28 +1,41 @@
...
 if [ "$PHP_FPM_USER_ID" != "" ]; then
-    (>&2 echo "Warning: PHP_FPM_USER_ID is deprecated. Please use WWW_DATA_USER_ID instead.")
-    WWW_DATA_USER_ID=$PHP_FPM_USER_ID
+    usermod -u $PHP_FPM_USER_ID nobody
 fi
 
-# Set the uid that www-data will run as if one was specified
-if [ "$WWW_DATA_USER_ID" != "" ]; then
-    usermod -u $WWW_DATA_USER_ID www-data
-fi
```
Comment (Chekote) [inline entrypoint.sh]: Why did you remove the `WWW_DATA_USER_ID` and restore the deprecated `PHP_FPM_USER_ID`?

File: entrypoint.sh [hunk 2]
```diff
@@ -1,28 +1,41 @@
-#!/usr/bin/env bash
+#!/usr/bin/env sh
 
 if [ "$PHP_FPM_USER_ID" != "" ]; then
-    (>&2 echo "Warning: PHP_FPM_USER_ID is deprecated. Please use WWW_DATA_USER_ID instead.")
-    WWW_DATA_USER_ID=$PHP_FPM_USER_ID
+    usermod -u $PHP_FPM_USER_ID nobody
```
Comment (Chekote) [inline entrypoint.sh]: Why did you change from modifying `www-data` to modifying `nobody`?

**Review by Chekote — CHANGES_REQUESTED** [review]

File: entrypoint.sh (hunk 1 above)
Comment (juanmrad) [reply entrypoint.sh]: my bad. I removed both while testing and forgot to add back.

File: entrypoint.sh (hunk 2 above)
Comment (juanmrad) [reply entrypoint.sh]: reverted.

**Comment by juanmrad** [comment]
while re-reviewing it I modified it a bit more to be more optimized ( avoid repetition ). I have also updated the documentation a bit more.

**Comment by Chekote** [comment]
It's not working. I rebuilt the PHP 5 images and ran this:

```
$ dcri! -e PHP_ALL_SESSION__SAVE_HANDLER=redis chekote/php:5 php -i | grep session.save_handler
session.save_handler => files => files
```
